Felienne/hedy
# highlighting of digits translations

Added coloring of digits according to the language translation.

**Description**

A new regex has been added to the file highlighting/definition.py to support the translation of the numbers.

This translation is then used in the definition of the numbers in the file highlighting/rules_list.py.

Concerning the recovery of the translation of the numbers, it could not be done like the classical commands, because we want a translation for all the digits.

Automated tests have been added for this new feature

**Fixes _issue or discussion number_**

fix #2835 

**How to test**

check that the front-end is not broken, and that the coloring of the Arabic numbers works

![image](https://user-images.githubusercontent.com/48509466/172782038-485719b0-7eb1-4e22-9e09-2a99fbe9a51b.png)

![image](https://user-images.githubusercontent.com/48509466/172782152-487f8927-1aba-446c-9465-8150ad3fd02a.png)

here are some arabic numbers (this will make the tests easier)
```
    0 : ٠            5 : ٥
    1 : ١            6 : ٦
    2 : ٢            7 : ٧
    3 : ٣            8 : ٨
    4 : ٤            9 : ٩
  283 : ٢٨٣        142 : ١٤٢
  255 : ٢٥٥        269 : ٢٦٩
  217 : ٢١٧        208 : ٢٠٨
  511 : ٥١١        587 : ٥٨٧
  423 : ٤٢٣        368 : ٣٦٨
22.52 : ٢٢.٥٢    28.66 : ٢٨.٦٦
38.72 : ٣٨.٧٢     30.2 : ٣٠.٢
48.66 : ٤٨.٦٦    26.38 : ٢٦.٣٨
30.08 : ٣٠.٠٨    48.54 : ٤٨.٥٤
 40.4 : ٤٠.٤     59.54 : ٥٩.٥٤
```
